src/urllib3/exceptions.py

```diff
@@ -32,9 +32,9 @@ class HTTPWarning(Warning):
 class PoolError(HTTPError):
     """Base exception for errors caused within a pool."""
 
-    pool: "ConnectionPool"
+    pool: Optional["ConnectionPool"]
```

Comment:
Preexisting: this line looks redundant to me, unless I'm missing something.

Reply:
hmmm, @sethmlarson Could you please elaborate on it?